catalog/solr/catalog-solr-offline-gazetteer/src/main/java/ddf/catalog/solr/offlinegazetteer/OfflineGazetteerPlugin.java

```diff
@@ -0,0 +1,206 @@
+
+  private final SolrClientFactory clientFactory;
+  private final SolrClient solrClient;
+
+  public OfflineGazetteerPlugin(SolrClientFactory clientFactory) {
+    this.clientFactory = clientFactory;
+    this.solrClient = clientFactory.newClient(STANDALONE_GAZETTEER_CORE_NAME);
+  }
+
+  @Override
+  public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+    List<Metacard> gazetteerMetacards =
+        input
+            .getCreatedMetacards()
+            .stream()
+            .filter(this::isGazetteerMetacard)
+            .collect(Collectors.toList());
+
+    if (gazetteerMetacards.isEmpty()) {
+      return input;
+    }
+
+    try {
+      solrClient.add(
+          STANDALONE_GAZETTEER_CORE_NAME,
+          gazetteerMetacards
+              .stream()
+              .map(OfflineGazetteerPlugin::convert)
+              .collect(Collectors.toList()));
+    } catch (SolrServerException | IOException e) {
+      LOGGER.debug("Error while processing gazetteer data", e);
+      throw new PluginExecutionException(e);
+    }
+
+    return input;
+  }
+
+  @Override
+  public UpdateResponse process(UpdateResponse input) throws PluginExecutionException {
+    List<Metacard> gazetteerMetacards =
+        input
+            .getUpdatedMetacards()
+            .stream()
+            .map(Update::getNewMetacard)
+            .filter(this::isGazetteerMetacard)
+            .collect(Collectors.toList());
+
+    if (gazetteerMetacards.isEmpty()) {
+      return input;
+    }
+
+    try {
+      solrClient.add(
+          STANDALONE_GAZETTEER_CORE_NAME,
+          gazetteerMetacards
+              .stream()
+              .map(OfflineGazetteerPlugin::convert)
+              .collect(Collectors.toList()));
+    } catch (SolrServerException | IOException e) {
+      LOGGER.debug("Error while processing gazetteer data", e);
+      throw new PluginExecutionException(e);
+    }
+    return input;
+  }
+
+  @Override
+  public DeleteResponse process(DeleteResponse input) throws PluginExecutionException {
+    List<String> ids =
+        input
+            .getDeletedMetacards()
+            .stream()
+            .filter(this::isGazetteerMetacard)
+            .map(Metacard::getId)
+            .collect(Collectors.toList());
+    if (ids.isEmpty()) {
+      return input;
+    }
+
+    try {
+      solrClient.deleteById(ids);
+    } catch (SolrServerException | IOException e) {
+      LOGGER.debug("Error while processing gazetteer data", e);
+      throw new PluginExecutionException(e);
+    }
+
+    return input;
+  }
+
+  protected static SolrInputDocument convert(Metacard metacard) {
+    SolrInputDocument solrDoc = new SolrInputDocument();
+    Consumer<String> getAttrAndAdd =
+        (attribute) ->
+            Optional.ofNullable(getStringAttribute(metacard, attribute))
+                .ifPresent(attr -> solrDoc.addField(attribute + "_txt", attr));
+
+    getAttrAndAdd.accept(Metacard.DESCRIPTION);
```

Comment:
I am not suggesting to hardcode catalog attributes for the plugin.  The catalog plugin can use the constants for attributes.  But it should then map those catalog attributes into gazetteer fields or call some gazetteer API to add the document that would know those field names.  By doing either of those, the `GazetteerQueryOfflineSolr` will not need to know anything about the catalog.  So for example, convert metacard title to gazetteer name.  Generally a gazetteer would not talk about titles.
https://www.geonames.org/5308655/phoenix.html

Reply:
I will join you in being overzealous then. 😎

Reply:
I did do the mapping but to your specific example of 'title' I can't change because that is the field the suggester is configured to make suggestions off of 😄

Reply:
Updated `title` to `name` now that the solrconfig has a separate suggester.